pkg/kubectl/scale.go

```diff
@@ -81,6 +82,17 @@ func (precondition *ScalePrecondition) Validate(controller *api.ReplicationContr
 	return nil
 }
 
+// ValidateJob ensures that the preconditions match.  Returns nil if they are valid, an error otherwise
+func (precondition *ScalePrecondition) ValidateJob(job *experimental.Job) error {
+	if precondition.Size != -1 && *job.Spec.Parallelism != precondition.Size {
```

Comment:
> Parallelism should never be nil, will always be set by defaults.

Thinking on it, I'd still prefer the protection since it's possible to create the object without going through conversions.  Has the argument been had in another pull?